README.md

```diff
@@ -0,0 +1,127 @@
+
+[![](https://jitpack.io/v/tillhub/scan-engine.svg)](https://jitpack.io/#tillhub/scan-engine)
+[![API](https://img.shields.io/badge/API-24%2B-green.svg?style=flat)](https://android-arsenal.com/api?level-11)
+# Scan Engine
+
+This library combines various scanning protocols into a single, easy-to-use interface.
+
+# How to setup
+
+**Step 1.** Add the JitPack repository to your `settings.gradle` file:
+
+```groovy
+dependencyResolutionManagement {
+    repositories {
+        ...
+        mavenCentral()
+		maven { url 'https://jitpack.io' }
+    }
+}
+```
+
+**Step 2.** Add the dependency to your app `build.gradle`:
+```groovy
+dependencies {
+    implementation 'com.github.tillhub:scan-engine:x.x.x'
+}
+```
+# Usage
+
+There are two ways you can use scan SDK:
+* `Camera scanner` : Manually register an activity or fragment result callback for the camera scanner.
+* `Barcode scanner` : Register a BroadcastReceiver with the barcode scanner.
+
+### 1. Camera Scanner (Activity/Fragment):
+
+*  `Singleton Access`: Use `@Inject lateinit var scanEngine: ScanEngine` to obtain a singleton reference to the Scan Engine instance.
+*  `Initialization`: Create a per Activity or Fragment Scanner instance using `scanEngine.newCameraScanner(this)`. This method automatically selects the appropriate scanner based on the device manufacturer (SunmiScanner or DefaultScanner).
+*  `Initiate Scan`: Call `scanner.startCameraScanner()` to initiate scanning. Pass appropriate scan key if needed, by default its null.
+*  `Handle Scan Results`: Subscribe to the `scanEngine.observeScannerResults()` flow to receive ScanEvent objects containing scan data.
+
+```kotlin
+
+@Inject lateinit var scanEngine: ScanEngine
+
+override fun onCreate(savedInstanceState: Bundle?) {
+    // ...
+
+    val scanner = scanEngine.newCameraScanner(this)
+    
+    scannerButton.setOnClickListener {
+        scanner.startCameraScanner()
+    }
+
+    // Observing scan result with scanEngine or scanner
+    scanEngine.observeScannerResults().collect { scanEvent ->
+         // Handle scanEvent result
+    }
+    // OR
+    scanner.observeScannerResults().collect { scanEvent ->
+        // Handle scanEvent result
+    }
+}
+```
+
+### 2. Barcode Scanner (BroadcastReceiver):
```

Comment:
I would remove the "BroadcastReceiver" here, it is unnecessarily precise. With future implementations of bluetooth scanners it may even become wrong.